api/src/test/java/org/pmiops/workbench/billing/FreeTierBillingServiceTest.java

```diff
@@ -55,221 +59,617 @@
+        testStartTime
+            .minus(Period.ofDays(limit / 2))
+            .minus(Period.ofDays(1))
+            // avoid silly test errors due to comparing too close to day boundaries
+            .minus(Duration.ofMinutes(1));
+    user.setFirstRegistrationCompletionTime(Timestamp.from(registrationTime));
+    userDao.save(user);
+
+    final Instant expirationTime = registrationTime.plus(Period.ofDays(limit));
+    final long daysRemaining = Duration.between(testStartTime, expirationTime).toDays();
+    final LocalDate expirationDate = expirationTime.atZone(ZoneId.systemDefault()).toLocalDate();
+
+    freeTierBillingService.checkFreeTierBillingUsage();
+    verify(notificationService)
+        .alertUserFreeTierTimeThreshold(
+            eq(user), eq(daysRemaining), eq(expirationDate), eq(dollarBalance));
+
+    // check that we do not alert twice for the 50% threshold
+
+    freeTierBillingService.checkFreeTierBillingUsage();
+    verifyZeroInteractions(notificationService);
+  }
+
+  @Test
+  public void checkFreeTierBillingUsage_exceeds75PctDayThreshold() {
+
+    // set cost values to ensure we don't alert from cost
+
+    final double dollarLimit = 100.0;
+    final double spent = 0.0;
+    final double dollarBalance = dollarLimit - spent;
+    workbenchConfig.billing.defaultFreeCreditsDollarLimit = dollarLimit;
+    doReturn(mockBQTableSingleResult(spent)).when(bigQueryService).executeQuery(any());
+
+    final DbUser user = createUser(SINGLE_WORKSPACE_TEST_USER);
+    createWorkspace(user, SINGLE_WORKSPACE_TEST_PROJECT);
+
+    final short limit = 1000;
+    workbenchConfig.billing.defaultFreeCreditsDaysLimit = limit;
+
+    final Instant testStartTime = Instant.now();
+
+    // check that we alert for the 75% threshold
+
+    // set my registration time to "more than 3/4 of my free credits ago"
+    final Instant registrationTime =
+        testStartTime
+            .minus(Period.ofDays(limit * 3 / 4))
+            .minus(Period.ofDays(1))
+            // avoid silly test errors due to comparing too close to day boundaries
+            .minus(Duration.ofMinutes(1));
+    user.setFirstRegistrationCompletionTime(Timestamp.from(registrationTime));
+    userDao.save(user);
+
+    final Instant expirationTime = registrationTime.plus(Period.ofDays(limit));
+    long daysRemaining = Duration.between(testStartTime, expirationTime).toDays();
+    final LocalDate expirationDate = expirationTime.atZone(ZoneId.systemDefault()).toLocalDate();
+
+    freeTierBillingService.checkFreeTierBillingUsage();
+    verify(notificationService)
+        .alertUserFreeTierTimeThreshold(
+            eq(user), eq(daysRemaining), eq(expirationDate), eq(dollarBalance));
+
+    // check that we do not alert twice for the 75% threshold
+
+    freeTierBillingService.checkFreeTierBillingUsage();
+    verifyZeroInteractions(notificationService);
+  }
+
+  @Test
+  public void checkFreeTierBillingUsage_exceeds100PctDayThreshold() {
```

Comment:
I think you can combine all of these tests like you did for the $ thresholds test if you pass in `Clock` to `FreeTierBillingService`.

AFAIK, this is a fairly common practice we use. You can take a look at `UserService` as an example. The current timestamp can then be fetched with the following `final Timestamp timestamp = new Timestamp(clock.instant().toEpochMilli());`.